lib/bundler/gem_helper.rb

```diff
@@ -98,7 +98,7 @@ def rubygem_push(path)
         allowed_push_host = @gemspec.metadata["allowed_push_host"]
         gem_command += " --host #{allowed_push_host}" if allowed_push_host
       end
-      unless allowed_push_host || Pathname.new("~/.gem/credentials").expand_path.file?
+      unless allowed_push_host || Bundler.user_home.join(".gem/credentials").file?
```

Comment:
Does `expand_path` raise if HOME isn't set? If it doesn't raise, I think we shouldn't create or use the temporary directory when looking for gem creds.